### ISSUE TITLE
Getting different dataset shape after preprocessing 

### ISSUE THREAD

**Issue by Tandon-A** [pclucas14/lidar_generation#5]
Hello

I read this paper recently and wanted to try my hands on it. Thank you for providing the code for the same.
I have one doubt with regards to the preprocessing step. 

The preprocessing step should return a tensor of shape (N, 60, 512, 3) but on trying the code, the tensor comes out to be of shape (N, 40, 256, 3)
Can you clarify whether it's okay to have this output tensor? Also, can you shed some light on the following two lines mentioned in the preprocessing function defined in utils.py

	Line 151: dataset = dataset[:, 5:45]
	Line 180: return dataset[:, :, :, ::2]

I think that line 151 is reducing the circles or clusters from 60 to 40 and then line 180 is reducing the point per circle/cluster to 256 from 512. 

If it's okay to have this as the output tensor, is there any reasoning for reducing clusters and points per cluster?

Please help. 

Thanks
Abhishek

**Comment by pclucas14** [comment]
Hi Abhishek, 

What you said is correct. I downsampled the input mainly to reduce the computational burden and improve training speed. I also discarded the outer circles because they usually are the ones that contain the most noise, and thus are the hardest to model. You can bypass these steps if you want, however you may need to make a few changes in the model architecture. 

Best, 
Lucas

**Comment by Tandon-A** [comment]
This clarifies it. Thank you for providing quick resolution.

**Comment by Tandon-A** [comment]
Hello Lucas, 

I had a small doubt regarding the visualization of the output of the GAN model. 

As per my understanding, The GAN model will also produce a tensor of shape 40 X 256 X 3, and then the show_pc function is being used to convert this output back to lidar form so that one can get the images as shown in the project's readme. 

I am not sure as to what's the use of line 205 in utils.py in the show_pc function. 
	Line 204: color = (velo[:, 2] - velo[:, 2].min()) / (velo[:, 2].max() - velo[:, 2].min())
	Line 205: color = (velo[:, 2] --0.069667026) / ( 0.0041348818--0.069667026)

Does line 204 need to be commented? And what do the values in line 205 signify?

Regards 
Abhishek

**Comment by pclucas14** [comment]
Hi, 

Yes the GAN will also produce 40 x 256 x 3 tensors (altough I recommend to use the polar representation for GANs to get nicer samples, which would be 40 x 256 x 2. You would then convert it back using `from_polar_np` to get 40 x 256 x 3). 

Line 204 and 205 do not actually modify the point cloud, _they only modify the display color for each point_ (hence the name `color`). Line 204 normalizes everything in `[0,1]`. I don't recall exactly what was the motivation for line 205. I did a quick test with line 205 commented out, and the colors were a little different. I think I just played around with values at line 205 until I got some colors I liked. 
So I would leave everything as is.

Hope this helps, 
Lucas
